daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/dsom/CompiledExpression1.scala

```diff
@@ -383,13 +392,61 @@ class DPathElementCompileInfo(
         matchesERD
       }
 
+    //
+    // We must indicate for all children having this path step as their name
```

Comment:
I'm kindof iffy on this. Technically, I'm not sure findNamedMatch is the right place for this, since there could be code that's trying to find a named match but for whatever reason they aren't used in an expression. I don't think that's the case, but the name doesn't really imply there will be side effects.

Perhaps just add a TODO to remove this when query-style expressions are supported? I think when query style expression are supported this function will return a Seq of matches, and then the caller can mark isReferencedByExpression as appropriate?

Reply:
Just fixing this to name it findNamedMatches and put the setting of the indicator var in a separate private method called at each place findNamedMatches is called.